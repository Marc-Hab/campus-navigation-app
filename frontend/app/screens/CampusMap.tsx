//Campus Map Screen
import { StyleSheet, Text, View, TextInput, Image } from "react-native";

import OutdoorMap from "../components/maps/OutdoorMap";
import { StatusBar } from "expo-status-bar";
import AppLogo from "../components/ui/AppLogo";
import { ThemeProvider } from "../components/context/ThemeContext";
import ThemeToggle from "../components/ui/ThemeToggle";
import useTheme from "../hooks/useTheme";
import { getStyles } from "../styles";

export default function CampusMap() {
  const { theme, toggleTheme } = useTheme(); // Use the custom hook
  const styles = getStyles(theme); // Get styles based on theme

  return (
    <View style={styles.container}>
      {/* Adding the status bar on top of the screen */}
      <View>
        <StatusBar style="dark" />
      </View>
      <View style={styles.iconRow}>
        <View style={styles.iconRowView}>
          <AppLogo />
        </View>
        <View>
          <Text style={(styles.iconRowView, styles.logo)}>Campus Pilot</Text>
        </View>
        {/*Dark/light theme toggle*/}
        <ThemeToggle />
      </View>

      {/*SearchBar*/}
      <View style={styles.searchBar}>
        <TextInput placeholder={"Search"} placeholderTextColor={"#666"} />
      </View>
      {/*Campus Map*/}
      <View style={styles.mapContainer}>
        <OutdoorMap />
      </View>
    </View>
  );
}
{
  /*Styles for the views*/
}
