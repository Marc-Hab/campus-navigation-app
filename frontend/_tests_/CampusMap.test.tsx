import React from "react";
import renderer from "react-test-renderer";
import CampusMap from "@/app/screens/OutdoorNav/CampusMap";

test("Campus Map renders correctly", () => {
  const tree = renderer.create(<CampusMap />).toJSON();
  expect(tree).toMatchSnapshot();
});
