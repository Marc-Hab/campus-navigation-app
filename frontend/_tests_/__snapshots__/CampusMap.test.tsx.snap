// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Campus Map renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flex": 0.7,
      }
    }
  >
    <View />
    <View
      style={
        {
          "alignContent": "space-around",
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "padding": 20,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "flex": 1,
          }
        }
      >
        <View>
          <Image
            source={
              {
                "testUri": "../../../assets/images/logo.png",
              }
            }
            style={
              {
                "height": 50,
                "width": 50,
              }
            }
            tintColor="#fff"
          />
        </View>
      </View>
      <View>
        <Text
          style={
            [
              {
                "alignItems": "center",
                "color": "#fff",
                "fontSize": 30,
                "fontWeight": "bold",
                "justifyContent": "space-around",
                "padding": 10,
              },
            ]
          }
        >
          Campus Pilot
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#000",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../assets/images/sleep-mode.png",
                }
              }
              style={
                {
                  "height": 40,
                  "width": 40,
                }
              }
              tintColor="#fff"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "position": "absolute",
        "top": 140,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#575757",
          "borderRadius": 10,
          "marginHorizontal": 10,
          "opacity": 1,
          "padding": 10,
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        SGW
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#AAAAAA",
          "borderRadius": 10,
          "marginHorizontal": 10,
          "opacity": 1,
          "padding": 10,
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Loyola
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "borderRadius": 10,
        "flex": 2,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    />
  </View>
</View>
`;
